### ISSUE TITLE
Android 8.0: App screen is half black after opening with open keyboard

### ISSUE THREAD

**Issue by jaanushansen** [codenameone/CodenameOne#2212]
Hi!

Testing Android 8.0 on Google Pixel introduced the following problem:

when app is reopened so that previously virtual keyboard was open, then CN1 uses wrong screen size and leaves lower part of view black. Full screen is used only after opening keyboard in app.

See attached flow:
[Android8.0AppReopened.pdf](https://github.com/codenameone/CodenameOne/files/1284512/Android8.0AppReopened.pdf)

Test app is based on Get Started App:
`
package mypackage;

import com.codename1.io.Log;
import com.codename1.ui.Button;
import com.codename1.ui.Command;
import com.codename1.ui.Display;
import com.codename1.ui.FontImage;
import com.codename1.ui.Form;
import com.codename1.ui.Dialog;
import com.codename1.ui.Label;
import com.codename1.ui.MenuBar;
import com.codename1.ui.TextField;
import com.codename1.ui.animations.CommonTransitions;
import com.codename1.ui.layouts.BorderLayout;
import com.codename1.ui.layouts.BoxLayout;
import com.codename1.ui.layouts.FlowLayout;
import com.codename1.ui.layouts.LayeredLayout;
import com.codename1.ui.plaf.UIManager;
import com.codename1.ui.util.Resources;
import com.codename1.ui.util.UITimer;
import java.io.IOException;
import com.codename1.ui.Toolbar;

public class TestApp {

    private Form current;
    private Resources theme;

    public void init(Object context) {
        theme = UIManager.initFirstTheme("/theme");
        Toolbar.setGlobalToolbar(true);
    }
    
    public void start() {
        if(current != null){
            current.show();
            return;
        }
        Form hi = new Form("Welcome", new BoxLayout(BoxLayout.Y_AXIS));
        hi.add(new Label("text:"));
        hi.add(new TextField());

        hi.addCommand(new Command("test1"));
        hi.addCommand(new Command("test2"));
        
        hi.show();
    }

    public void stop() {
        current = Display.getInstance().getCurrent();
        if(current instanceof Dialog) {
            ((Dialog)current).dispose();
            current = Display.getInstance().getCurrent();
        }
    }
    
    public void destroy() {
    }

}
`

**Comment by jaanushansen** [comment]
Info from debugging:

Added debug log to [Display](https://github.com/codenameone/CodenameOne/blob/master/CodenameOne/src/com/codename1/ui/Display.java) sizeChanged method, when app is reopened (screen 1.3 -> 1.4):

`
09-07 14:58:09.654 called Display.sizeChanged(w = 1080, h = 990)
09-07 14:58:09.654 call addSizeChangeEvent: 1080, 990
09-07 14:58:09.654 called Display.sizeChanged(w = 1080, h = 1731)
09-07 14:58:09.654 (w == current.getWidth() && h == current.getHeight()) == true, return without adding event
09-07 14:58:09.656 called Form.sizeChangedInternal(w = 1080, h = 990)
`

that is currently is not handled the case, when size changing events come in so quickly, that the previous event is not yet handled.

**Comment by DurankGts** [comment]
This same issue it occur some time in android 6

**Comment by jaanushansen** [comment]
Extra debugging did introduce one another problematic place:

Added debug to [AndroidAsyncView](https://github.com/codenameone/CodenameOne/blob/master/Ports/Android/src/com/codename1/impl/android/AndroidAsyncView.java) onSizeChanged method:

        System.out.println("Display.getInstance().callSerially(" + w + ", " + h + ")");
        Display.getInstance().callSerially(new Runnable() {
            public void run() {
                System.out.println("handleSizeChange(" + w + ", " + h  + ")");
                cn1View.handleSizeChange(w, h);
            }
        });

It do not happen often, but 1 out of 10 retries the log is:
```
09-07 15:40:19.484 Display.getInstance().callSerially(1080, 990)
09-07 15:40:19.524 Display.getInstance().callSerially(1080, 1731)
09-07 15:40:19.536 handleSizeChange(1080, 1731)
09-07 15:40:19.572 handleSizeChange(1080, 990)
```

That is sometimes code in EDT is called in mixed order and as result form gets height 990, but not the latest height 1731

**Comment by jaanushansen** [comment]
Hi! Probably you fixed the race in Display (or maybe something go wrong on rotation?), but what about [the race in AndroidAsyncView](https://github.com/codenameone/CodenameOne/issues/2212#issuecomment-327790826) ?

**Comment by codenameone** [comment]
I don't see the race potential there but maybe it's something I'm missing. Either way it's a separate issue
